enterprise/server/util/cpuset/cpuset_test.go

```diff
@@ -104,3 +104,60 @@ func TestCPUSetDisabledManualCPUSet(t *testing.T) {
 	defer cancel1()
 	assert.Equal(t, []int{0, 1, 3}, cpus1)
 }
+
+func TestNumaNodeFairness(t *testing.T) {
+	flags.Set(t, "executor.cpu_leaser.enable", true)
+	flags.Set(t, "executor.cpu_leaser.overhead", 0)
+	flags.Set(t, "executor.cpu_leaser.min_overhead", 0)
+	flags.Set(t, "executor.cpu_leaser.cpuset", "0:0-3,1:4-7")
```

Comment:
If the parser can handle it, shuffling up the cpu indexes a bit more would give us higher confidence that the NUMA node is being taken into account, instead of e.g. just returning the first N cpus. (Same for the tests below)

```suggestion
	flags.Set(t, "executor.cpu_leaser.cpuset", "0:0-1,1:4-7,0:2-3")
```

Reply:
Done. (yuck!)